### PR TITLE
Collect private network info when docker containers are exposed to hosts

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

- Collect network docker container network info even when exposed to the host
- Extend unit tests

### Motivation

Provide more container network info, used by the network tracer

### CODE REVIEW

File: pkg/util/docker/containers.go [hunk 1]
```diff
@@ -209,14 +209,14 @@ func parseContainerNetworkAddresses(ports []types.Port, netSettings *types.Summa
...
-		} else {
-			// Port is not exposed, cache it and retieve IPs from network settings
-			tempAddrList = append(tempAddrList, containers.NetworkAddress{
-				Port:     int(port.PrivatePort),
-				Protocol: port.Type,
-			})
 		}
+		// Cache container ports
+		tempAddrList = append(tempAddrList, containers.NetworkAddress{
+			Port:     int(port.PrivatePort),
```
Comment (leeavital) [inline pkg/util/docker/containers.go]: Is this intentionally missing an IP ?

File: pkg/util/docker/docker_test.go [hunk 2]
```diff
@@ -92,86 +92,337 @@ func TestResolveImageName(t *testing.T) {
...
+						IPv6Gateway:         "",
+						GlobalIPv6Address:   "",
+						GlobalIPv6PrefixLen: 0,
+						MacAddress:          "MacAddress",
+					},
+				},
+			},
+			expected: []containers.NetworkAddress{
+				{
+					IP:       net.ParseIP("0.0.0.0"),
```
Comment (leeavital) [inline pkg/util/docker/docker_test.go]: I thoufght what we discussed in this case was getting the host IP an not 0.0.0.0

Comment (ahmed-mez) [reply pkg/util/docker/docker_test.go]: But if it's not mentioned, docker default the value it to `0.0.0.0`. Do you suggest not reporting it if it's `0.0.0.0` ?

File: pkg/util/docker/containers.go (hunk 1 above)
Comment (ahmed-mez) [reply pkg/util/docker/containers.go]: added in the next loop, from `netSettings.Networks`, the logic won't send tuples without IPs.

File: pkg/util/docker/docker_test.go (hunk 2 above)
Comment (ahmed-mez) [reply pkg/util/docker/docker_test.go]: @leeavital I think we can mark this as resolved based on our discussion

**Review by xlucas — APPROVED** [review]
LGTM

File: pkg/util/docker/containers.go (hunk 1 above)
Comment (leeavital) [inline pkg/util/docker/containers.go]: Is there guaranteed to be a pricate port, or will we need to check for 0  ?

**Review by leeavital — APPROVED** [review]

**Pull request merged**